trading_calendars/exchange_calendar_asex.py

```diff
@@ -0,0 +1,168 @@
+#
+# Copyright 2019 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time, timedelta
+from itertools import chain
+import pandas as pd
+from pandas.tseries.holiday import (
+    EasterMonday,
+    GoodFriday,
+    Holiday,
+)
+from pytz import timezone, UTC
+
+from .common_holidays import (
+    assumption_day,
+    christmas,
+    christmas_eve,
+    epiphany,
+    european_labour_day,
+    new_years_day,
+    orthodox_easter,
+)
+from .trading_calendar import (
+    TradingCalendar,
+    HolidayCalendar,
+)
+
+
+NewYearsDay = new_years_day()
+
+Epiphany = epiphany()
+
+OrthodoxAshMonday = orthodox_easter() - timedelta(48)
+
+NationalHoliday1 = Holiday(
+    'National Holiday 1',
+    month=3,
+    day=25,
+)
+
+OrthodoxGoodFriday = orthodox_easter() - timedelta(2)
+
+OrthodoxEasterMonday = orthodox_easter() + timedelta(1)
+
+LabourDay = european_labour_day()
+
+OrthodoxWhitMonday = orthodox_easter() + timedelta(50)
+
+AssumptionDay = assumption_day()
+
+NationalHoliday2 = Holiday(
+    'National Holiday 2',
+    month=10,
+    day=28,
+)
+
+ChristmasEve = christmas_eve(start_date='2009')
+
+ChristmasDay = christmas()
+
+SecondDayOfChristmas = Holiday(
+    'Second Day of Christmas (w/ no added Friday off)',
+    month=12,
+    day=26,
+)
+
+
+class ASEXExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Athens Stock Exchange (ASEX).
+
+    Open Time: 10:00 AM, EEST
+    Close Time: 5:25 PM, EEST
```

Comment:
I'm not sure where 5:25 came from (I just saw it on the wiki), I'll ask Jamie about it

Reply:
Checked with Jamie, for now we are gonna go with open at 10:00, close at 5:00 until Sept 29, 2008, after which close becomes 5:25

Reply:
Sorry, I'm still confused if it's 5:25 or 5:20 😆 You've defined it below as 5:20. Also, if going by [this](https://www.reuters.com/article/greece-exchange/greek-stock-exchange-to-extend-trading-hours-idUSL1887378920080718), is the open time not 10:30?

Reply:
Oops forgot to change the comments!  Fixing now.  And yeah, Jamie thinks it might've gotten moved to 10:00 at some point but he's not sure when, so just use 10:00 for now.